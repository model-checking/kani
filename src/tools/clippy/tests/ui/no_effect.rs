#![feature(box_syntax)]
#![warn(clippy::no_effect_underscore_binding)]
#![allow(dead_code)]
#![allow(path_statements)]
#![allow(clippy::deref_addrof)]
#![allow(clippy::redundant_field_names)]
#![feature(untagged_unions)]

struct Unit;
struct Tuple(i32);
struct Struct {
    field: i32,
}
enum Enum {
    Tuple(i32),
    Struct { field: i32 },
}
struct DropUnit;
impl Drop for DropUnit {
    fn drop(&mut self) {}
}
struct DropStruct {
    field: i32,
}
impl Drop for DropStruct {
    fn drop(&mut self) {}
}
struct DropTuple(i32);
impl Drop for DropTuple {
    fn drop(&mut self) {}
}
enum DropEnum {
    Tuple(i32),
    Struct { field: i32 },
}
impl Drop for DropEnum {
    fn drop(&mut self) {}
}
struct FooString {
    s: String,
}
union Union {
    a: u8,
    b: f64,
}

fn get_number() -> i32 {
    0
}
fn get_struct() -> Struct {
    Struct { field: 0 }
}
fn get_drop_struct() -> DropStruct {
    DropStruct { field: 0 }
}

unsafe fn unsafe_fn() -> i32 {
    0
}

fn main() {
    let s = get_struct();
    let s2 = get_struct();

    0;
    s2;
    Unit;
    Tuple(0);
    Struct { field: 0 };
    Struct { ..s };
    Union { a: 0 };
    Enum::Tuple(0);
    Enum::Struct { field: 0 };
    5 + 6;
    *&42;
    &6;
    (5, 6, 7);
    box 42;
    ..;
    5..;
    ..5;
    5..6;
    5..=6;
    [42, 55];
    [42, 55][1];
    (42, 55).1;
    [42; 55];
    [42; 55][13];
    let mut x = 0;
    || x += 5;
    let s: String = "foo".into();
    FooString { s: s };
    let _unused = 1;
    let _penguin = || println!("Some helpful closure");
    let _duck = Struct { field: 0 };
    let _cat = [2, 4, 6, 8][2];

    #[allow(clippy::no_effect)]
    0;

    // Do not warn
    get_number();
    unsafe { unsafe_fn() };
    let _used = get_struct();
    let _x = vec![1];
    DropUnit;
    DropStruct { field: 0 };
    DropTuple(0);
    DropEnum::Tuple(0);
    DropEnum::Struct { field: 0 };
}
