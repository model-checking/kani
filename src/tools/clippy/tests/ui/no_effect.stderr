error: statement with no effect
  --> $DIR/no_effect.rs:65:5
   |
LL |     0;
   |     ^^
   |
   = note: `-D clippy::no-effect` implied by `-D warnings`

error: statement with no effect
  --> $DIR/no_effect.rs:66:5
   |
LL |     s2;
   |     ^^^

error: statement with no effect
  --> $DIR/no_effect.rs:67:5
   |
LL |     Unit;
   |     ^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:68:5
   |
LL |     Tuple(0);
   |     ^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:69:5
   |
LL |     Struct { field: 0 };
   |     ^^^^^^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:70:5
   |
LL |     Struct { ..s };
   |     ^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:71:5
   |
LL |     Union { a: 0 };
   |     ^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:72:5
   |
LL |     Enum::Tuple(0);
   |     ^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:73:5
   |
LL |     Enum::Struct { field: 0 };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:74:5
   |
LL |     5 + 6;
   |     ^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:75:5
   |
LL |     *&42;
   |     ^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:76:5
   |
LL |     &6;
   |     ^^^

error: statement with no effect
  --> $DIR/no_effect.rs:77:5
   |
LL |     (5, 6, 7);
   |     ^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:78:5
   |
LL |     box 42;
   |     ^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:79:5
   |
LL |     ..;
   |     ^^^

error: statement with no effect
  --> $DIR/no_effect.rs:80:5
   |
LL |     5..;
   |     ^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:81:5
   |
LL |     ..5;
   |     ^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:82:5
   |
LL |     5..6;
   |     ^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:83:5
   |
LL |     5..=6;
   |     ^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:84:5
   |
LL |     [42, 55];
   |     ^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:85:5
   |
LL |     [42, 55][1];
   |     ^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:86:5
   |
LL |     (42, 55).1;
   |     ^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:87:5
   |
LL |     [42; 55];
   |     ^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:88:5
   |
LL |     [42; 55][13];
   |     ^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:90:5
   |
LL |     || x += 5;
   |     ^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:92:5
   |
LL |     FooString { s: s };
   |     ^^^^^^^^^^^^^^^^^^^

error: binding to `_` prefixed variable with no side-effect
  --> $DIR/no_effect.rs:93:5
   |
LL |     let _unused = 1;
   |     ^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::no-effect-underscore-binding` implied by `-D warnings`

error: binding to `_` prefixed variable with no side-effect
  --> $DIR/no_effect.rs:94:5
   |
LL |     let _penguin = || println!("Some helpful closure");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: binding to `_` prefixed variable with no side-effect
  --> $DIR/no_effect.rs:95:5
   |
LL |     let _duck = Struct { field: 0 };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: binding to `_` prefixed variable with no side-effect
  --> $DIR/no_effect.rs:96:5
   |
LL |     let _cat = [2, 4, 6, 8][2];
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 30 previous errors

